models/settings_helpers.go

```diff
@@ -42,6 +46,22 @@ func GetSettings(q reform.DBTX) (*Settings, error) {
 		return nil, errors.Wrap(err, "failed to unmarshal settings")
 	}
 
+	// Extract UUID from default PMM PostgreSQL service ID to use it as an unique identifier for the PMM server.
+	// Service id example: /service_id/78fae40e-9dc8-4efd-a40c-972a208dd218
+	service, err := FindServiceByName(q, DefaultPMMPostgreSQLService)
+	if err == nil {
+		// the service exists
+		idParts := strings.Split(service.ServiceID, "/")
+		if len(idParts) < 3 {
+			return nil, errors.Errorf("wrong format of service ID of defalt PMM PostgreSQL service: number of id parts is %d", len(idParts))
+		}
+		s.PMMServerID = idParts[len(idParts)-1] // uuid is at the end
+	} else {
+		// The default service will be added after first start of PMM Server, we should not return error if we fail to get it.
+		ctx := logger.Set(q.Context(), "models")
+		logger.Get(ctx).Errorf("Failed to get default PMM PostgreSQL service by name %q: %s", DefaultPMMPostgreSQLService, err)
+	}
```

Comment:
Why not just generate UUID on the server startup?

Reply:
+1

Reply:
there are discussions about the uuid in 2 places  
https://perconacorp.slack.com/archives/GJAGH01PD/p1637598896051600
https://perconacorp.slack.com/archives/GV9GJJY77/p1637596006135100?thread_ts=1637595529.134800&cid=GV9GJJY77

Reply:
but even i am not sure why we are re-using the postgres service ID, even if we want to have 2 separate uuids for the server and telemetry, why don't we generate a server ID on startup and store it in the DB and use it everytime, i think that should get rid of all this extra logic

Reply:
I don't know, we just grabbed what seemed unique. I can change it to the logic you proposed.

Reply:
@maxkondr your thoughts?

Reply:
I would say that it's better to generate a new one and store it in settings model instead of trying to reuse already exist service ID. We may decide to remove PostgreSQL inside PMM in future or remove it from monitoring, in that case we will be in trouble.

Reply:
Done, PMM Server ID is generated separately on startup.